src/routes/transactions/mappers/multisig-transactions/multisig-transaction-details.mapper.ts

```diff
@@ -115,33 +114,23 @@ export class MultisigTransactionDetailsMapper {
             ),
         );
 
-    const promises: Promise<unknown>[] = [];
-    promises.push(
-      gasToken != NULL_ADDRESS
-        ? this.tokenRepository.getToken(chainId, gasToken)
-        : Promise.resolve(null),
-    );
-    promises.push(
-      transaction.executor
-        ? this.addressInfoHelper.getOrDefault(chainId, transaction.executor, [
-            'CONTRACT',
-          ])
-        : Promise.resolve(null),
-    );
-    promises.push(
-      this.addressInfoHelper.getOrDefault(
-        chainId,
-        transaction.refundReceiver ?? NULL_ADDRESS,
-        ['CONTRACT'],
-      ),
-    );
-    promises.push(
-      txStatus === TransactionStatus.Cancelled
-        ? this._getRejectors(chainId, transaction, safe)
-        : Promise.resolve(null),
-    );
     const [gasTokenInfo, executor, refundReceiver, rejectors] =
-      await Promise.all(promises);
+      await Promise.all([
+        gasToken != NULL_ADDRESS
+          ? this.tokenRepository.getToken(chainId, gasToken)
+          : Promise.resolve(null),
+        transaction.executor
+          ? this.addressInfoHelper.getOrDefault(chainId, transaction.executor, [
+              'CONTRACT',
+            ])
+          : Promise.resolve(null),
+        this.addressInfoHelper.getOrDefault(
+          chainId,
+          transaction.refundReceiver ?? NULL_ADDRESS,
+          ['CONTRACT'],
+        ),
+        this._getRejectors(chainId, transaction, safe),
+      ]);
```

Comment:
I moved these promises inside the `Promise.all` to preserve the types so we no longer need cast.